### PR TITLE
8274069: Clean up g1ParScanThreadState a bit

### PR DESCRIPTION
**Pull request by tschatzl**
Hi all,

  can I have reviews for this change that does some minor refactoring for issues found in JDK-8271880/PR#5037:

- remove declaration without definition
- move inlined code in the .hpp file into .inline.hpp file
- factor out some code

Testing: gha, local compilation, local gc/g1

Thanks,
  Thomas

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8274069](https://bugs.openjdk.java.net/browse/JDK-8274069): Clean up g1ParScanThreadState a bit

### Reviewers
 * [Stefan Johansson](https://openjdk.java.net/census#sjohanss) (@kstefanj - **Reviewer**) ⚠️ Review applies to 53997bce9abeedbd6b9d27011eabd52267bdf933
 * [Albert Mingkun Yang](https://openjdk.java.net/census#ayang) (@albertnetymk - **Reviewer**) ⚠️ Review applies to 8cf9b6a9bb775f8cc6afa9bb5abad0a787428706
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5607/head:pull/5607` \
`$ git checkout pull/5607`

Update a local copy of the PR: \
`$ git checkout pull/5607` \
`$ git pull https://git.openjdk.java.net/jdk pull/5607/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5607`

View PR using the GUI difftool: \
`$ git pr show -t 5607`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5607.diff">https://git.openjdk.java.net/jdk/pull/5607.diff</a>

</details>

### CODE REVIEW

**Review by kstefanj — APPROVED** [review]
👍

**Review by albertnetymk — APPROVED** [review]
The old name, `enqueue_card_after_barrier_filters`, is still referenced in the comments.

(I also wonder if it makes sense to unify all callers of `enqueue_card_if_tracked` to use `write_ref_field_post` with template non-type arguments. Not in *this* PR ofc.)

File: src/hotspot/share/gc/g1/g1ParScanThreadState.hpp [hunk 1]
```diff
@@ -130,28 +128,16 @@ class G1ParScanThreadState : public CHeapObj<mtGC> {
 
   void push_on_queue(ScannerTask task);
 
-  template <class T> void enqueue_card_if_tracked(G1HeapRegionAttr region_attr, T* p, oop o) {
-    assert(!HeapRegion::is_in_same_region(p, o), "Should have filtered out cross-region references already.");
-    assert(!_g1h->heap_region_containing(p)->is_young(), "Should have filtered out from-young references already.");
+  // Apply the post barrier to the given reference field. Enqueues the card of p
+  // if the barrier (same region, not from survivor) does not filter out not find
```
Comment (kimbarrett) [inline src/hotspot/share/gc/g1/g1ParScanThreadState.hpp:132]: I don't understand this comment; seems like there's a grammar or wording problem.

**Review by kimbarrett — CHANGES_REQUESTED** [review]
Does this change allow this inclusion of an .inline.hpp to be moved out of the .hpp?
33 #include "gc/g1/heapRegionRemSet.inline.hpp"

**Comment by tschatzl** [comment]
> Does this change allow this inclusion of an .inline.hpp to be moved out of the .hpp?
> 33 #include "gc/g1/heapRegionRemSet.inline.hpp"

In the latest change I also touched up includes in that file a bit. Nothing about heapRegionRemSet has been used in that file...

**Review by kimbarrett — APPROVED** [review]
Looks good.

**Comment by tschatzl** [comment]
Thanks @albertnetymk @kstefanj @kimbarrett for your reviews.

/integrate

**Pull request closed without merging**